projects/Mallard/src/screens/auth-switcher-screen.tsx

```diff
@@ -0,0 +1,296 @@
+import React, { useState, useCallback, useEffect } from 'react'
+import {
+    View,
+    TextInput,
+    Text,
+    KeyboardAvoidingView,
+    Keyboard,
+} from 'react-native'
+import {
+    fetchAndPersistUserAccessTokenWithIdentity,
+    fetchMembershipDataForKeychainUser,
+    canViewEdition,
+} from 'src/authentication/helpers'
+import { facebookAuthWithDeepRedirect } from 'src/authentication/services/facebook'
+import { googleAuthWithDeepRedirect } from 'src/authentication/services/google'
+import { Button, ButtonAppearance } from 'src/components/button/button'
+import { appAppearances } from 'src/theme/appearance'
+import { metrics } from 'src/theme/spacing'
+import { TitlepieceText } from 'src/components/styled-text'
+import { FadeIn } from 'src/components/bounce-fade-in'
+import { Spinner } from 'src/components/spinner'
+
+enum AuthStatus {
+    pending = 0,
+    authed = 1, // can't guarantee the callback will navigate away so will leave this here
+    unauthed = 2,
+    authenticating = 3,
+}
+
+const useRandomState = () =>
+    useState(
+        Math.random()
+            .toString()
+            .split('.')[1],
+    )[0]
+
+const tryAuth = async (
+    authPromise: Promise<unknown>,
+    {
+        onStart = () => {},
+        onSuccess,
+        onError,
+    }: {
+        onStart?: () => void
+        onSuccess: () => void
+        onError: (err: string) => void
+    },
+) => {
+    try {
+        await authPromise
+        onStart()
+        const membershipData = await fetchMembershipDataForKeychainUser()
```

Comment:
How often will this get called? We should be aiming for daily with some spread within an hour.

Reply:
It only gets called once on auth ... for now